pkg/util/kubernetes/kubelet/podwatcher.go

```diff
@@ -41,8 +42,11 @@ func NewPodWatcher(expiryDuration time.Duration) (*PodWatcher, error) {
 		kubeUtil:       kubeutil,
 		lastSeen:       make(map[string]time.Time),
 		lastSeenReady:  make(map[string]time.Time),
-		tagsDigest:     make(map[string]string),
```

Comment:
I feel like it is simpler to always initialise the `tagsDigest` map. 
it will remove 2: `if isWatchingTags { ...` for "creation" and "deletion". I think that, because you never rely on the fact that the `map == nil`, instead you set the boolean `isWatchingTags `

Another solution could be:
- removing `isWatchingTags` and instead `if isWatchingTags {` do `if watcher.tagsDigest != nil {`

Reply:
it would remove 2 `if isWatchingTags` but add 2 `if watcher.tagsDigest != nil` or I am missing something?

Reply:
I think `isWatchingTags` is equal to `watcher.tagsDigest != nil`
So yes, it still need to have `if watcher.tagsDigest != nil {` in several places.

so mabye a cleaner solution can be:
```golang
func (w *PodWatcher) isWatchingTags() bool {
     return w.tagsDigest != nil 
}
```

Reply:
any way it is just cosmetic...